iroh/src/magicsock.rs

```diff
@@ -3238,22 +3238,19 @@ mod tests {
 
         // For each node, start a task which monitors its local endpoints and registers them
         // with the other nodes as local endpoints become known.
-        let mut tasks = JoinSet::new();
+        let mut tasks = Vec::new();
```

Comment:
Wait?  What?  A `JoinSet` aborts all tasks on drop already.  Why are we jumping through all these hoops?

Reply:
bonus fun 🤣 I can go back to joinset I guess